apps/src/applab/AppLabView.jsx

```diff
@@ -21,50 +21,76 @@ class AppLabView extends React.Component {
   static propTypes = {
     handleVersionHistory: PropTypes.func.isRequired,
     autogenerateML: PropTypes.func.isRequired,
+    isEditingProject: PropTypes.bool.isRequired,
+    screenIds: PropTypes.arrayOf(PropTypes.string).isRequired,
+    onScreenCreate: PropTypes.func.isRequired,
+    onMount: PropTypes.func.isRequired,
+
+    // Provided by redux
     hasDataMode: PropTypes.bool.isRequired,
     hasDesignMode: PropTypes.bool.isRequired,
     interfaceMode: PropTypes.oneOf([
       ApplabInterfaceMode.CODE,
       ApplabInterfaceMode.DESIGN,
       ApplabInterfaceMode.DATA
     ]).isRequired,
-    isEditingProject: PropTypes.bool.isRequired,
-
-    screenIds: PropTypes.arrayOf(PropTypes.string).isRequired,
-    onScreenCreate: PropTypes.func.isRequired,
-
-    onMount: PropTypes.func.isRequired
+    isRtl: PropTypes.bool,
+    widgetMode: PropTypes.bool
   };
 
   componentDidMount() {
     this.props.onMount();
   }
 
   render() {
-    const codeWorkspaceVisible =
-      ApplabInterfaceMode.CODE === this.props.interfaceMode;
+    const {
+      interfaceMode,
+      widgetMode,
+      isRtl,
+      isEditingProject,
+      screenIds,
+      onScreenCreate,
+      autogenerateML,
+      hasDesignMode,
+      hasDataMode,
+      handleVersionHistory
+    } = this.props;
```

Comment:
Just a side-note question here out of my own personal curiosity - is this part of our style guide or just preference based?

Reply:
our style guide doesn't have specific preferences like this (though it should), but this has become an unofficial part of it (e.g., you'll see it commonly throughout frontend code, and folks will leave this as a nit comment on PRs)